src/settings/NotifTroubleshootingScreen.js

```diff
@@ -0,0 +1,499 @@
+  const alerts = [];
+  report.problems.forEach(problem => {
+    switch (problem) {
+      case NotificationProblem.SystemSettingsDisabled:
+        alerts.push(
+          <AlertItem
+            buttons={[
+              {
+                id: 'fix',
+                label: 'Open settings',
+                onPress: () => {
+                  openSystemNotificationSettings();
+                },
+              },
+            ]}
+            bottomMargin
+            text="Notifications are disabled in system settings."
+          />,
+        );
+        break;
+
+      case NotificationProblem.GooglePlayServicesNotAvailable:
+        // TODO: Could offer as a solution in case the user is actually fine
+        //   with Google Play Services and just has a problem with its setup:
+        //   - It looks like you can ask Android to run a native flow to
+        //     fix problems with Google Play Services:
+        //       https://developers.google.com/android/reference/com/google/android/gms/common/GoogleApiAvailability
+        alerts.push(<AlertItem bottomMargin text="Notifications require Google Play Services." />);
+        break;
+
+      case NotificationProblem.ServerHasNotEnabled:
+        alerts.push(
+          <AlertItem
+            bottomMargin
+            text={
+              isAtLeastAdmin
+                ? {
+                    text: 'The Zulip server at {realm} is not set up to deliver push notifications. Please contact your administrator.',
+                    values: { realm: identity.realm.toString() },
+                  }
+                : {
+                    text: 'The Zulip server at {realm} is not set up to deliver push notifications.',
+                    values: { realm: identity.realm.toString() },
+                  }
+            }
+            buttons={[
+              {
+                id: 'learn-more',
+                label: 'Learn more',
+                onPress: () => {
+                  openLinkWithUserPreference(
+                    new URL(
+                      'https://zulip.readthedocs.io/en/stable/production/mobile-push-notifications.html',
+                    ),
+                    settings,
+                  );
+                },
+              },
+            ]}
+          />,
+        );
+        break;
+
+      case NotificationProblem.TokenNotAcked: {
+        // TODO: Could offer:
+        //   - Re-request the device token from the platform (#5329 may be
+        //     an obstacle), and show if a/the request hasn't completed
+        //   - Re-request registering token with server, and show if a/the
+        //     request hasn't completed
+        alerts.push(genericAlert);
+      }
+    }
+  });
+
+  return (
+    <Screen scrollEnabled={false} title="Troubleshooting" padding>
+      <SafeAreaView mode="padding" edges={['right', 'left']} style={styles.contentArea}>
+        {
+          uniq(alerts) // Deduplicate genericAlert, which multiple problems might map to
```

Comment:
The enum might have distinct values that we want to include in `report.problems` (since that comes to us in support requests) but that we want to map to the same user-facing alert, the alert that you should contact support. When that's the case, it would be buggy to show multiple identical "contact support" alerts when just one would do, and so we want to deduplicate them.

In the current revision, testing whether an alert is one of the ones to deduplicate is simple: is it (triple-equals) `genericAlert`, or not? It's true that we'd have a problem, when building `alerts`, if we meant an alert to be included in deduplication but we failed to use `genericAlert` for it. My thinking here was that as long as we didn't make that mistake, it wouldn't matter how complex the alerts were.

I can write something that avoids pushing multiple `genericAlert`s to `alerts` in the first place, so that we don't have to apply `uniq` to `alerts` on the way to rendering the UI. But if we deduplicate `report.problems` itself, then we'll potentially lose information in the support-request payload.

Reply:
> The enum might have distinct values that we want to include in report.problems (since that comes to us in support requests) but that we want to map to the same user-facing alert, the alert that you should contact support.

Er, stepping back a bit, I guess the enum doesn't currently have such values. So maybe we just add a TODO for if it eventually does?

Reply:
OK, I see why deduping the list of `NotificationProblem`s wouldn't necessarily be the right direction.

If we end up coming back and adding more complexity here, I'd still want to avoid comparing these complex UI-describing objects. This would be a fine solution, as long as `genericAlert` is the only one we want to dedupe:

> I can write something that avoids pushing multiple `genericAlert`s to `alerts` in the first place, so that we don't have to apply `uniq` to `alerts` on the way to rendering the UI.